flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -958,6 +958,12 @@ private String convertToString(Object o) {
 		} else if (o.getClass() == Duration.class) {
 			Duration duration = (Duration) o;
 			return String.format("%d ns", duration.toNanos());
+		} else if (o instanceof List) {
+			return String.join(";", ((List<?>) o).stream().map(Object::toString).toArray(String[]::new));
```

Comment:
Could you also escape the entries in the list? Parsing lists supports escaping with `'` character.
So that e.g. A;'A;B';C is parsed as ["A", "A;B", "C"]. The same applies for Maps. I think when writing values out it's safe to always escape them.

You can check some examples of escaping in `StructuredOptionsSplitterTest` or in `ReadableWritableConfigurationTest` see test for `list<string>`.

Reply:
Why isn't this using `Collectors.joining(";")`?
i.e.,
```
return ((List<?>) o).stream().map(this::convertToString).collect(Collectors.joining(";"));
```

Reply:
I will escape the value string when converting `List` and `Map` to structured string. Also `Collectors.joining(";")` will make the codes more concise.